Homework02/Kursovik/C_sharp_Kursovik_OOP/PathLine.cs

```diff
@@ -0,0 +1,376 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace C_sharp_Kursovik_OOP
+{
+    public static class PathLine
+    {
+        /// <summary>
+        /// Показать все диски
+        /// </summary>
+        /// <returns></returns>
+        public static List<DriveInfo> GetDrives()
+        {
+            DriveInfo[] allDrives = DriveInfo.GetDrives();
+            List<DriveInfo> newAllDrives = new List<DriveInfo>();
+            foreach (var driveInfo in allDrives)
+            {
+                if (driveInfo.IsReady)
+                {
+                    newAllDrives.Add(driveInfo);
+                }
+            }
+            return newAllDrives;
+        }
+        /// <summary>
+        /// Показать все файлы
+        /// </summary>
+        /// <param name="path"></param>
+        /// <returns></returns>
+        public static List<FileInfo> GetFiles(String path)
+        {
+            if (path == "" || path == null)
+            {
+                return null;
+            }
+            if (Path.HasExtension(path))
+            {
+                path = Path.GetDirectoryName(path);
+            }
+            return new DirectoryInfo(path).GetFiles().ToList();
+        }
+        /// <summary>
+        /// Показать все папки и файлы в заданной директории
+        /// </summary>
+        /// <param name="dirName"></param>
+        public static void GetShow(string dirName)
+        {
+            Console.WriteLine();
+            if (Directory.Exists(dirName))
+            {
+                Console.WriteLine("Подкаталоги:");
+                string[] dirs = Directory.GetDirectories(dirName);
+                foreach (string s in dirs)
+                {
+                    Console.WriteLine(s);
+                }
+                Console.WriteLine();
+                Console.WriteLine("Файлы:");
+                string[] files = Directory.GetFiles(dirName);
```

Comment:
```suggestion
                var files = Directory.EnumerateFiles(dirName);
```
Если можно не создавать массив, то нужно НЕ(!) создавать массив...